### PR TITLE
fix[DVM2.0]: address pendingStakes issue in voting contract

### PR DESCRIPTION
**Pull request by chrismaree**
**Motivation**

While writing health checks for the dvm we discovered a small issue in the voterdapp that can cause slashing trackers to diverge from their expected values. This occurs in a very specific case: when a user stakes additional tokens during an active reveal, before revealing their vote.

This PR both shows the bug (the test added here would fail if the but still existed) and shows the fix. 

Note that I had to remove a low impact require message to keep this contracts bytecode below the max size.

**Summary**

Shows a bug in the voting contract and fixes it

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [X]  New unit tests created
- []  Existing tests adequate, no new tests required
- [X]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/core/test/hardhat/data-verification-mechanism/VotingV2.js [hunk 1]
```diff
@@ -3988,6 +3988,81 @@ describe("VotingV2", function () {
     const finalContractBalance = await votingToken.methods.balanceOf(voting.options.address).call();
     assert.equal(finalContractBalance, "0");
   });
+  it("Staking during active reveal is dealt with correctly via pendingStakes", async function () {
+    // An issue was found in the contract wherein a staked voter stakes again during the active reveal round, thereby
+    // increasing their effective stake to a value higher than what they had at the variable freeze time (first commit).
```
Comment (md0x) [inline packages/core/test/hardhat/data-verification-mechanism/VotingV2.js:3993]:
```suggestion
    // increasing their effective stake to a value higher than what they had at the variable freeze time (first commit or stake during active reveal phase).
```

In the test case the first variable freeze occurs with the stake

**Review by md0x — APPROVED** [review]
Good job finding the issue @chrismaree! The solution makes perfect sense.

**Review by mrice32 — APPROVED** [review]
LGTM! Nice find and fix!

**Review by md0x — APPROVED** [review]
LGTM

**Pull request merged**